packages/pwa-kit-create-app/scripts/trim-extensions.js

```diff
@@ -72,191 +73,115 @@ function trimExtensions(directory, generatedPlugins) {
 }
 
 function processFile(filePath, plugins) {
+    let modified = false // flag to indicate if the file was modified
+    let blockMarkers = [] // stack of block markers
+    let removedBlocks = [] // list of blocks that were removed
+    let skippingBlock = false // flag to indicate if we are skipping a block
+
     // Read source file
     const source = fs.readFileSync(filePath, 'utf-8')
 
     // Search file content for plugin references
-    const pluginPositions = []
     const pluginRegex = new RegExp(Object.keys(plugins).join('|'), 'g')
-    let match
-    while ((match = pluginRegex.exec(source)) !== null) {
-        pluginPositions.push({
-            start: match.index,
-            end: match.index + match[0].length,
-            name: match[0] // store which plugin was found
-        })
-    }
-    if (pluginPositions.length === 0) {
-        // No plugins found in file, return early
-        return false
-    }
-
-    // Parse into AST
-    const ast = parser.parse(source, {
-        sourceType: 'module',
-        plugins: ['jsx', 'typescript']
-    })
-
-    // Track if we made any changes
-    let modified = false
-
-    // Shared helper to evaluate boolean expressions with plugins
-    const evaluateLogicalExpression = (node) => {
-        if (node.type === 'Identifier' && Object.keys(plugins).includes(node.name)) {
-            return plugins[node.name]
-        }
-        if (node.type === 'LogicalExpression') {
-            const left = evaluateLogicalExpression(node.left)
-            const right = evaluateLogicalExpression(node.right)
-            if (node.operator === '&&') {
-                return left && right
+    if (pluginRegex.test(source)) {
```

Comment:
@kzheng-sfdc this is still using the optimization we discussed around not doing per-line regex, right? I think that's what `line 85` (unchanged) does?

Reply:
It's no longer needed as we only need a file-wide check.